data/tutorials/language/1ds_01_arrays.md

```diff
@@ -0,0 +1,158 @@
+
+```ocaml
+# let even_numbers = Array.init 5 (fun i -> i * 2);;
+val even_numbers : int array = [|0; 2; 4; 6; 8|]
+```
+
+## Accessing Array Elements
+
+You can access individual elements of an array using the `.(index)` syntax, with the index of the element you want to access. The index starts from 0 and goes up to the size of the array minus one.  For example, to access the third element of an array `even_numbers`, you would write:
+
+```ocaml
+# even_numbers.(2);;
+- : int = 4
+```
+
+## Modifying Array Elements
+
+To modify an element in an array, we simply assign a new value to it using the indexing operator. For example, to change the value of the third element of the array `even_numbers` created above to 42, we have to write:
+
+```ocaml
+# even_numbers.(2) <- 42;;
+- : unit = ()
+```
+
+Note that this modification doesn’t return the modified array: the value returned by this operation is `unit`, `even_numbers` is modified in place in the current scope. Updating an array's content is a side effect.
+
+## The Standard Library `Array` Module
+
+OCaml provides several useful functions for working with arrays. Here are some of the most common ones:
+
+### Length of an Array
+
+The `Array.length` function returns the size of an array. For example, to get the size of the array `even_numbers` created above:
+
+```ocaml
+# Array.length even_numbers;;
+- : int = 5
+```
+
+### Iterate on an Array
+
+The `Array.iter` function applies a given function to each element of an array. For example, to print all the elements of the array `zeroes` created above, we can apply the `print_int` (as this is an `int array`) function to each element:
+
+```ocaml
+# Array.iter (fun x -> print_int x; print_string " ") zeroes;;
+1 2 42 4 5 - : unit = ()
+```
+
+### Map an Array
+
+The `Array.map` function applies a given function to each element of an array and returns a new array with the results. For example, to create a new array that contains the square of each element of the array `even_numbers` created above, we can apply the following square function `fun x -> x * x` to each element of the array:
+
+```ocaml
+# Array.map (fun x -> x * x) even_numbers;;
+- : int array = [|0; 4; 1764; 36; 64|]
+```
+
+### Folding an Array
+
+To combine all the elements of an array into a single result, we can use the `Array.fold_left` and `Array.fold_right` functions. These functions take a binary function, an initial accumulator value, and an array as arguments. The binary function takes two arguments: the current value of the accumulator and the current element of the array and returns a new accumulator value. Both functions traverse the array
+ but in opposite directions. This is essentially the same as `List.fold_left` and `List.fold_right`.
+
+Here is the signature of `Array.fold_left`:
+
+```ocaml
+val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b array -> 'a
```

Comment:
```suggestion
# Array.fold_left;;
val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b array -> 'a = <fun>
```
I also get the `= <fun>` at the end of the result. Plus, I'd put the prompt command in there for clarity.